### PR TITLE
Implemented Starts for intervals

### PR DESCRIPTION
**Pull request by jbradl11**
This PR implements the "Starts" operator for Intervals:

From the the 1.2 Spec:

Description:
The starts operator returns true if the first interval starts the second. More precisely, if the starting point of the first is equal to the starting point of the second interval and the ending point of the first interval is less than or equal to the ending point of the second interval.
This operator uses the semantics described in the Start and End operators to determine interval boundaries.
If precision is specified and the point type is a date/time type, comparisons used in the operation are performed at the specified precision.
If either argument is null, the result is null.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository. 
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Code coverage has not gone down and all code touched or added is covered.
- [x] All dependent libraries are appropriately updated or have a corresponding PR related to this change

Regression Test passes and matches expected results for Bonnie-v2.1

**Reviewer: c-monkey**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

**Comment by c-monkey** [comment]
Successfully tested with the revised measure made by @mayerm94 from https://oncprojectracking.healthit.gov/support/browse/BONNIE-442. No longer has issues with Start or End operators.

File: test/elm/interval/data.cql [hunk 1]
```diff
@@ -654,6 +654,36 @@ define Foo: start of Interval[DateTime(2012, 1, 1), DateTime(2013, 1, 1)]
...
+define IntegerIntervalStartsFalse: Interval[1, 10] starts Interval[4, 10]
+define IntegerIntervalStartEndsFalse: Interval[4, 10] starts Interval[4, 9]
+define DecimalIntervalStartsTrue: Interval[4.0, 10.0] starts Interval[4.0, 15.0]
+define DecimalIntervalStartsFalse: Interval[1.0, 10.0] starts Interval[4.0, 10.0]
+define DecimalIntervalStartsEndsFalse: Interval[4.0, 10.0] starts Interval[4.0, 9.0]
+define QuantityIntervalStartsTrue: Interval[5.0 'g', 10.0 'g'] starts Interval[5.0 'g', 15.0 'g']
+define QuantityIntervalStartsFalse: Interval[1.0 'g', 10.0 'g'] starts Interval[5.0 'g', 10.0 'g']
+define QuantityIntervalStartsEndsFalse: Interval[5.0 'g', 10.0 'g'] starts Interval[5.0 'g', 9.9 'g']
+define DateTimeIntervalStartsTrue: Interval[DateTime(2012, 1, 5, 0, 0, 0, 0), DateTime(2012, 1, 25, 0, 0, 0, 0)] starts Interval[DateTime(2012, 1, 5, 0, 0, 0, 0), DateTime(2012, 1, 27, 0, 0, 0, 0)]
+define DateTimeIntervalStartsFalse: Interval[DateTime(2012, 1, 5), DateTime(2012, 1, 25)] starts day of Interval[DateTime(2012, 1, 6), DateTime(2012, 1, 27)]
```
Comment (c-monkey) [inline test/elm/interval/data.cql]: What do you think about adding a True test for "starts day of" as well?

File: src/datatypes/interval.coffee [hunk 2]
```diff
@@ -197,6 +197,22 @@ module.exports.Interval = class Interval
     catch
       false
 
+  starts: (other, precision) ->
+    if precision? and @low instanceof DateTime
+      startEqual = @low.sameAs(other.low, precision)
+    else
+      startEqual = cmp.equals(@low, other.low)
+    lessThanOrEqual = cmp.lessThanOrEquals(@high, other.high, precision)
```
Comment (c-monkey) [inline src/datatypes/interval.coffee]: I suggest changing this variable name to follow same convention as in ends, e.g. endLessThanOrEqual. Obviously this is just for symmetry and ease of maintenance.

**Comment by c-monkey** [comment]
Regressions improved:
```
hmdmlantana
====Patients Passing Regression===========
=: CMS ID: CMS31v7 RESULTS ARE EQUAL (33 == 33)
=: CMS ID: CMS111v7 RESULTS ARE EQUAL (28 == 28)
=: CMS ID: CMS55v7 RESULTS ARE EQUAL (21 == 21)
=: CMS ID: CMS32v8 RESULTS ARE EQUAL (36 == 36)
=: CMS ID: CMS53v7 RESULTS ARE EQUAL (55 == 55)
=: Population: Population Criteria Section  RESULTS ARE EQUAL (26 == 26)
=: Population: Stratification 1  RESULTS ARE EQUAL (27 == 27)
=: Population: Stratification 2  RESULTS ARE EQUAL (27 == 27)
=: Population: Population Criteria Section -0 RESULTS ARE EQUAL (19 == 19)
=: Population: Stratification 1 -1 RESULTS ARE EQUAL (20 == 20)
=: Population: Stratification 2 -2 RESULTS ARE EQUAL (20 == 20)
=: Population: Population Criteria Section -3 RESULTS ARE EQUAL (26 == 26)
=: Population: Stratification 1 -4 RESULTS ARE EQUAL (30 == 30)
=: Population: Stratification 2 -5 RESULTS ARE EQUAL (34 == 34)
+: Population: Stratification 3  SECOND BRANCH HAS MORE PASSES (24 < 33)
Measures with no differences: 14/15

====Code Coverage Percentage Regression===========
+: CMS ID: CMS111v7-0 SECOND BRANCH HAS MORE COVERAGE (98 < 100)
+: CMS ID: CMS55v7-0 SECOND BRANCH HAS MORE COVERAGE (98 < 100)
+: CMS ID: CMS32v8-0-1-2 SECOND BRANCH HAS MORE COVERAGE (82 < 100)
Measures with no differences: 9/12
```

**Comment by c-monkey** [comment]
I don't have permissions to check my boxes, will you please do so @jbradl11 ?
